### PR TITLE
edit quest details activity

### PR DESCRIPTION
**Pull request by isabelleleeson**
TODO in next pr: 
- send a request to the server when click save button
- remove mocked data and fetch real values
- send delete quest request to the server when click delete

image

### CODE REVIEW

File: android/app/src/main/java/com/chorestory/templates/RegisterRequest.java [hunk 1]
```diff
@@ -5,11 +5,13 @@
...
     private String name;
     private String password;
+    private int picture;
 
     public RegisterRequest(String email, String clanName, String name, String password) {
         this.email = email;
         this.clan_name = clanName;
         this.name = name;
         this.password = password;
+        this.picture = 3; // TODO
```
Comment (isabelleleeson) [inline android/app/src/main/java/com/chorestory/templates/RegisterRequest.java]: someone should make this fix in a different pr

File: android/app/src/main/java/com/chorestory/app/ParentQuestDetailsActivity.java [hunk 2]
```diff
@@ -1,18 +1,210 @@
...
+                R.array.recurrence_array, R.layout.spinner_item);
+        recurrenceSpinnerAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        selectRecurrenceSpinner.setAdapter(recurrenceSpinnerAdapter);
+        // Todo: fetch actual value for the recurrence string
+        selectRecurrenceSpinner.setSelection(recurrenceSpinnerAdapter.getPosition("Repeats monthly"));
+        selectRecurrenceSpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @Override
+            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+                recurrence = (String) parent.getItemAtPosition(position);
+                System.out.println("onItemSelected");
```
Comment (jimzhaozhengwang) [inline android/app/src/main/java/com/chorestory/app/ParentQuestDetailsActivity.java]: remove all `println`s in this file

File: android/app/src/main/java/com/chorestory/app/ParentQuestDetailsActivity.java [hunk 3]
```diff
@@ -1,18 +1,210 @@
...
+                final Calendar c = Calendar.getInstance();
+                int year = c.get(Calendar.YEAR);
+                int month = c.get(Calendar.MONTH);
+                int day = c.get(Calendar.DAY_OF_MONTH);
+
+                DatePickerDialog.OnDateSetListener listener = new DatePickerDialog.OnDateSetListener() {
+                    @Override
+                    public void onDateSet(DatePicker view, int year, int month, int dayOfMonth) {
+                        mDay = dayOfMonth;
+                        mMonth = month;
```
Comment (jimzhaozhengwang) [inline android/app/src/main/java/com/chorestory/app/ParentQuestDetailsActivity.java]: month is represented by an integer from 0 to 11
do `mMonth = month + 1` here instead of line 203
to reduce the number of places you need to do this adjustment as you still need to make a request with mMonth

File: android/app/src/main/java/com/chorestory/app/ParentQuestDetailsActivity.java [hunk 4]
```diff
@@ -1,18 +1,210 @@
...
+            @Override
+            public void onClick(View view) {
+                onBackPressed();
+            }
+        });
+    }
+
+    private String getDateText() {
+        // months used are 0-11
+        return mDay + "-" + (mMonth + 1) + "-" + mYear;
```
Comment (jimzhaozhengwang) [inline android/app/src/main/java/com/chorestory/app/ParentQuestDetailsActivity.java]: just use mMonth

File: android/app/src/main/java/com/chorestory/app/ParentQuestDetailsActivity.java (hunk 3 above)
Comment (isabelleleeson) [reply android/app/src/main/java/com/chorestory/app/ParentQuestDetailsActivity.java]: I want to keep mMonth as it is so I have the correct value when I want to use it to send to server

Comment (isabelleleeson) [reply android/app/src/main/java/com/chorestory/app/ParentQuestDetailsActivity.java]: nvm changed it
